Common/Securities/SecurityDefinitionSymbolResolver.cs

```diff
@@ -0,0 +1,163 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using QuantConnect.Configuration;
+using QuantConnect.Data.Auxiliary;
+using QuantConnect.Interfaces;
+using QuantConnect.Logging;
+using QuantConnect.Util;
+
+namespace QuantConnect.Securities
+{
+    /// <summary>
+    /// Resolves standardized security definitions such as FIGI, CUSIP, ISIN, SEDOL into
+    /// a properly mapped Lean <see cref="Symbol"/>.
+    /// </summary>
+    public class SecurityDefinitionSymbolResolver
+    {
+        private readonly List<SecurityDefinition> _securityDefinitions;
+        private readonly MapFileResolver _mapFileResolver;
+        
+        /// <summary>
+        /// Creates an instance of the symbol resolver
+        /// </summary>
+        /// <param name="dataFolder">Data folder to read the security-database.csv file from</param>
+        /// <param name="mapFileProvider">Map file provider used to obtain symbol mappings</param>
+        /// <param name="market">The market of the assets referred to by the security definitions</param>
+        public SecurityDefinitionSymbolResolver(
+            DirectoryInfo dataFolder = null,
+            IMapFileProvider mapFileProvider = null,
+            string market = Market.USA)
+        {
+            var dataFolderPath = dataFolder?.FullName ?? Globals.DataFolder;
+            
+            var securityDefinitionFile = new FileInfo(
+                Path.Combine(
+                    dataFolderPath,
+                    "symbol-properties",
+                    "security-database.csv"));            
+            
+            if (!SecurityDefinition.TryFromCsvFile(securityDefinitionFile, out _securityDefinitions))
+            {
+                Log.Error($"SecurityDefinitionSymbolResolver(): No security definitions data loaded from file: {securityDefinitionFile.FullName}");
+            }
+
+            if (mapFileProvider == null)
+            {
+                var dataProvider = Composer.Instance.GetExportedValueByTypeName<IDataProvider>(Config.Get("data-provider", "QuantConnect.Lean.Engine.DataFeeds.DefaultDataProvider"));
+                mapFileProvider = Composer.Instance.GetExportedValueByTypeName<IMapFileProvider>(Config.Get("map-file-provider", "LocalDiskMapFileProvider"));
+                mapFileProvider.Initialize(dataProvider);
+            }
+            
+            _mapFileResolver = mapFileProvider.Get(market);
```

Comment:
Don't think we need to hard code the market, we can just get it from the SID from the `SecurityDefinition` and determine the `mapFileResolver` at runtime 👍